src/graph/sampling/neighbor/neighbor.cc

```diff
@@ -104,6 +95,39 @@ HeteroSubgraph SampleNeighbors(
   return ret;
 }
 
+HeteroSubgraph SampleNeighbors(
+    const HeteroGraphPtr hg,
+    const std::vector<IdArray>& nodes,
+    const std::vector<int64_t>& fanouts,
+    EdgeDir dir,
+    const std::vector<FloatArray>& prob,
+    const std::vector<IdArray>& exclude_edges,
+    bool replace) {
+
+  // sanity check
+  CHECK_EQ(nodes.size(), hg->NumVertexTypes())
+    << "Number of node ID tensors must match the number of node types.";
+  CHECK_EQ(fanouts.size(), hg->NumEdgeTypes())
+    << "Number of fanout values must match the number of edge types.";
+  CHECK_EQ(prob.size(), hg->NumEdgeTypes())
+    << "Number of probability tensors must match the number of edge types.";
+
+  if (!exclude_edges.empty()) {
+    std::vector<IdArray> remain_edges(hg->NumEdgeTypes());
+    for (dgl_type_t etype = 0; etype < hg->NumEdgeTypes(); ++etype) {
+     EdgeArray edges = hg->Edges(etype);
+     remain_edges[etype] =
+            edges.excludeCertainEids<decltype(hg->NumEdges(etype))>(exclude_edges[etype]);
```

Comment:
`NumEdges` is always `int64_t`.  So this function will fail if the graph is `int32`.

You can dispatch to different primitive types using `ATEN_ID_TYPE_SWITCH` (similarly there are other switches).  It works like this:
```C++
ATEN_ID_TYPE_SWITCH(hg->DataType(), IdType, {
  remain_edges[etype] =
      edges.excludeCertainEids<IdType>(exclude_edges[etype]);
});
```

Reply:
Done in newest commit version.